src/core/instance/state.js

```diff
@@ -273,6 +273,12 @@ function initMethods (vm: Component, methods: Object) {
           `Avoid defining component methods that start with _ or $.`
         )
       }
+      if (methods[key] != null && typeof methods[key] !== 'function') {
+        warn(
+          `Method "${key}" should be a function.`
+        )
+        break
```

Comment:
you could instead set the method[key] to null, so the code works but does nothing